### ISSUE TITLE
Missed data if logs message is huge

### ISSUE THREAD

**Issue by yuriydzobak** [VictoriaMetrics/victorialogs-datasource#80]
Hi,
I'm using plugin version 0.5.0 and victorialogs 0.33.0 i've found an issue in grafana plugin
As you see iam able to see log entries in VMUI
![screenshot-000442](https://github.com/user-attachments/assets/e40e64c6-a09b-4779-abb2-e9fe56c7c911)

If i'm doing the same in grafana, it random show me the data, the limit range is 1k and time range 2 days too, if i select 1 hour limit it has the same problem
![screenshot-000443](https://github.com/user-attachments/assets/99926d3f-dd63-4a63-9cd4-d6dada8c4cf5)

```
_time:2d pod_namespace:monitoring container_name:lf | count()
{"count(*)":"1054"}
```
it has logs 1k entries

If i press button to run query it random show me logs in timeline each time

https://github.com/user-attachments/assets/5144e52b-fed6-4d22-8171-d31d5b91bc08

I beelive the problem im my logs message size
grafana show another logs entries without probelm, but that app provided the message size almost 200kb data in single message
VictroaLogs has message in logs like this
```
{"ts":"2024-10-05T08:38:55.780Z","level":"info","caller":"VictoriaMetrics/app/vlselect/main.go:161","msg":"client has canceled the request after 0.064 seconds: remoteAddr=\"10.10.94.253:59254\", requestURI: \"/select/logsql/query?end=1728117535&limit=1000&query=_time%3A2d+pod_namespace%3Amonitoring+container_name%3Alf+&start=1727944735\""}
{"ts":"2024-10-05T08:38:56.642Z","level":"info","caller":"VictoriaMetrics/app/vlselect/main.go:161","msg":"client has canceled the request after 0.075 seconds: remoteAddr=\"10.10.94.253:59270\", requestURI: \"/select/logsql/query?end=1728117536&limit=1000&query=_time%3A2d+pod_namespace%3Amonitoring+container_name%3Alf+&start=1727944736\""}
{"ts":"2024-10-05T08:38:57.337Z","level":"info","caller":"VictoriaMetrics/app/vlselect/main.go:161","msg":"client has canceled the request after 0.121 seconds: remoteAddr=\"10.10.94.253:59276\", requestURI: \"/select/logsql/query?end=1728117537&limit=1000&query=_time%3A2d+pod_namespace%3Amonitoring+container_name%3Alf+&start=1727944737\""}
{"ts":"2024-10-05T08:38:57.960Z","level":"info","caller":"VictoriaMetrics/app/vlselect/main.go:161","msg":"client has canceled the request after 0.065 seconds: remoteAddr=\"10.10.94.253:59280\", requestURI: \"/select/logsql/query?end=1728117537&limit=1000&query=_time%3A2d+pod_namespace%3Amonitoring+container_name%3Alf+&start=1727944737\""}
{"ts":"2024-10-05T08:38:58.734Z","level":"info","caller":"VictoriaMetrics/app/vlselect/main.go:161","msg":"client has canceled the request after 0.135 seconds: remoteAddr=\"10.10.94.253:59286\", requestURI: \"/select/logsql/query?end=1728117538&limit=1000&query=_time%3A2d+pod_namespace%3Amonitoring+container_name%3Alf+&start=1727944738\""}
{"ts":"2024-10-05T08:38:59.388Z","level":"info","caller":"VictoriaMetrics/app/vlselect/main.go:161","msg":"client has canceled the request after 0.066 seconds: remoteAddr=\"10.10.94.253:59296\", requestURI: \"/select/logsql/query?end=1728117539&limit=1000&query=_time%3A2d+pod_namespace%3Amonitoring+container_name%3Alf+&start=1727944739\""}
{"ts":"2024-10-05T08:39:00.186Z","level":"info","caller":"VictoriaMetrics/app/vlselect/main.go:161","msg":"client has canceled the request after 0.068 seconds: remoteAddr=\"10.10.94.253:56138\", requestURI: \"/select/logsql/query?end=1728117539&limit=1000&query=_time%3A2d+pod_namespace%3Amonitoring+container_name%3Alf+&start=1727944739\""}
{"ts":"2024-10-05T08:39:10.624Z","level":"info","caller":"VictoriaMetrics/app/vlselect/main.go:161","msg":"client has canceled the request after 0.109 seconds: remoteAddr=\"10.10.94.253:51920\", requestURI: \"/select/logsql/query?end=1728117549&limit=1000&query=_time%3A2d+pod_namespace%3Amonitoring+container_name%3Alf+%7C+unpack_json&start=1727944749\""}
{"ts":"2024-10-05T08:39:11.517Z","level":"info","caller":"VictoriaMetrics/app/vlselect/main.go:161","msg":"client has canceled the request after 0.105 seconds: remoteAddr=\"10.10.94.253:51932\", requestURI: \"/select/logsql/query?end=1728117551&limit=1000&query=_time%3A2d+pod_namespace%3Amonitoring+container_name%3Alf+%7C+unpack_json&start=1727944751\""}
```
In browser inspecttor, i don't see any error for grafana or network and so on issue

**Comment by hagen1778** [comment]
@dmitryk-dk is this related to https://github.com/VictoriaMetrics/victorialogs-datasource/issues/91 ?

**Comment by dmitryk-dk** [comment]
> @dmitryk-dk is this related to #91 ?

It looks like yes because the logs with 200kb can be parsed correctly by Scanner, we can ask @yuriydzobak to check my [PR](https://github.com/VictoriaMetrics/victorialogs-datasource/pull/95)  on his installation if it is possible. :)

**Comment by yuriydzobak** [comment]
> > @dmitryk-dk is this related to #91 ?
> 
> It looks like yes because the logs with 200kb can't be parsed correctly by Scanner, we can ask @yuriydzobak to check my [PR](https://github.com/VictoriaMetrics/victorialogs-datasource/pull/95) on his installation if it is possible. :)

yes, i can,

**Comment by dmitryk-dk** [comment]
> > > @dmitryk-dk is this related to #91 ?
> > 
> > 
> > It looks like yes because the logs with 200kb can't be parsed correctly by Scanner, we can ask @yuriydzobak to check my [PR](https://github.com/VictoriaMetrics/victorialogs-datasource/pull/95) on his installation if it is possible. :)
> 
> yes, i can, just provide me info where i can plugin to download or it will be some in next release

it is easy to run it from the branch. Just follow the next steps:
from the root of the repository
1. make victorialogs-frontend-plugin-build
2. make victorialogs-backend-plugin-build

you will see the folder with collected files 

after just making these steps 
https://github.com/VictoriaMetrics/victorialogs-datasource?tab=readme-ov-file#installation

and you can run Grafana on your local

**Comment by hagen1778** [comment]
https://github.com/VictoriaMetrics/victorialogs-datasource/pull/95 has been merged and will be available in the next release.

**Comment by dmitryk-dk** [comment]
The issue should be resolved in the released [v0.6.2](https://github.com/VictoriaMetrics/victorialogs-datasource/releases/tag/v0.6.2) version. @yuriydzobak thank you for the test.
